### DISCUSSION TITLE
Are there minimal NodeJS version, free of JS/TS dependencies?

### DISCUSSION THREAD

**Discussion by Ashvith10** [nodejs/help#4664] [Q&A] [answered]
I am trying to re-package NodeJS and every of it's dependencies separately on Guix to allow cache and prevent rebuilds, and after coming across a few JS/TS dependencies (in my case, undici), I can't help but ask if there exist any such version of NodeJS, that does not come with any dependencies, excluding C/C++ libraries?

**Comment by avivkeller** [comment, marked as answer]
Certain dependencies, such as corepack, NPM, and a few others, can be excluded from the build via CLI flags during the make process, but other than that, there is no minimal version of Node.js.

**Reply by Ashvith10** [reply]
What about `undici `and `npm`? Is there no way to get over circular dependencies?

**Reply by avivkeller** [reply]
Undici isn't optional, as it's used internally. NPM however (IIRC) be disabled when building, but I don't recall the exact steps to do so

**Reply by Ashvith10** [reply]
@RedYetiDev so I would have to bootstrap older version of NodeJS to build undici, to further build modern versions of NodeJS? What about node-slim - does that have non-JS dependencies?

**Reply by avivkeller** [reply]
I'm not sure about node-slim, please reach out to the developers of that for information on that.
